orm/auto_uint64.go

```diff
@@ -1,20 +1,21 @@
 package orm
 
 import (
+	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 var _ Indexable = &AutoUInt64TableBuilder{}
 
 // NewAutoUInt64TableBuilder creates a builder to setup a AutoUInt64Table object.
-func NewAutoUInt64TableBuilder(prefixData byte, prefixSeq byte, storeKey sdk.StoreKey, model Persistent) *AutoUInt64TableBuilder {
+func NewAutoUInt64TableBuilder(prefixData byte, prefixSeq byte, storeKey sdk.StoreKey, model codec.ProtoMarshaler, cdc codec.Marshaler) *AutoUInt64TableBuilder {
```

Comment:
Can we replace `codec.ProtoMarshaler` with `proto.Message`? That would be more clear, maybe for a separate PR though.

Reply:
I guess that would require changing the `Marshaler` interface upstream. So not relevant for this PR, but we should do that upstream.

Reply:
How would we replace `codec.ProtoMarshaler` with `proto.Message` in the `Marshaler` methods since `codec.ProtoMarshaler`'s methods like `Marshal()` seems to be needed (eg in `ProtoCodec`)? We would need to add type assertions in the `BinaryMarshaler` methods like it's the case in the `JSONMarshaler` methods.

Reply:
We would just use the `proto.Marshal` method instead which is the official  approach.

Reply:
If you look at https://github.com/regen-network/regen-ledger/blob/marie/136-x-group/proto/regen/group/v1alpha1/tx.proto#L18-L25, you can see that we have multiple methods for updating groups. I thought it would be better to have just one method `UpdateGroup` instead of `UpdateGroupMembers`, `UpdateGroupAdmin`, etc.

Reply:
https://github.com/cosmos/cosmos-sdk/issues/8412